### DISCUSSION TITLE
Some question about "msfile” and "bamboo(z)"

### DISCUSSION THREAD

**Discussion by hongtaoliuxing** [neperfepx/neper#626] [Q&A] [answered]
Hello everyone.
I want to do a layered model, divided into three layers.
The commands I use are as follows:
```
"3::msfile(3myfile)" -morpho "bamboo(z)"
```
The question is can the height of the layer be specified ?
For example, the total height is 1mm, the upper and lower height is 0.2mm, and the middle height is 0.6mm.
How to achieve ?
Thanks.

**Comment by rquey** [comment, marked as answer]
One way is to use `lamellar` instead of `bamboo`:
```
neper -T -n from_morpho -morpho "lamellar(w=0.2:0.6,pos=start,v=(0,0,1))" -o d
neper -V d.tess -print d
```

![image](https://user-images.githubusercontent.com/2659376/210366476-85cb2d46-0efe-4d67-810c-b42aeaa9615d.png)

**Reply by hongtaoliuxing** [reply]
Thank you for your answer, I am sorry to reply so late.
I used this command before, but I can 't subdivide the grains further.
The commands are as follows:
neper -T -n 3::1000 from_morpho -morpho "lamellar(w=0.2:0.6,pos=start,v=(0,0,1))" -domain "cube(4,2,1)"
neper -T -n from_morpho -morpho "lamellar(w=0.2:0.6,pos=start,v=(0,0,1))" -domain "cube(4,2,1)" "3::msfile(3myfile)"
neper -T -n from_morpho -morpho "lamellar(w=0.2:0.6,pos=start,v=(0,0,1))" -domain "cube(4,2,1)" -o 3
neper -T -loadtess 3.tess "3::msfile(3myfile)"
These are Bad arguments!
It should be that I am not proficient enough about neper.
How should I do to achieve it？
Thank you again for your reply.
